bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliSpec.java

```diff
@@ -44,58 +44,61 @@
 public class CliSpec {
     private final Option option;
     private final Operand[] operands;
-    private Object[] mainArgs;
+    private final List<Object> mainArgs;
     private final String[] args;
 
-    public CliSpec(Option option, Operand[] operands, String[] args) {
+    public CliSpec(Option option, Operand[] operands, String... args) {
         //Todo: refactor code and write tests
         this.option = option;
         this.operands = operands;
         this.args = args;
+        mainArgs = new ArrayList<>();
     }
 
     public Object[] getMainArgs() {
+        // First argument is for the strand
+        mainArgs.add(null);
         if (option != null) {
             BMap<BString, Object> record = option.parseRecord(args);
-            int size = (operands.length + 1) * 2 + 1;
-            mainArgs = new Object[size];
             processOperands(option.getOperandArgs());
-            mainArgs[size - 2] = record;
-            mainArgs[size - 1] = true;
+            mainArgs.add(record);
+            mainArgs.add(true);
         } else {
-            mainArgs = new Object[operands.length * 2 + 1];
-            // Todo: create an empty option and call option.getOperandArgs()
             Module module = new Module("$anon", ".", "0.0.0");
```

Comment:
We can make this dummy module a constant. IMO the correct way would be to pass the root module from codegen to CLispec and use that. ( We already done that in config passing)

Reply:
Fixed by passsing `null` for module as this is a dummy record